packages/neon_framework/lib/src/widgets/image.dart

```diff
@@ -100,7 +101,10 @@ class NeonCachedImage extends StatefulWidget {
       return cached;
     }
 
-    final data = await getImage();
+    var data = await getImage();
+    try {
+      data = utf8.encode(ImageUtils.rewriteSvgDimensions(utf8.decode(data)));
+    } catch (_) {}
```

Comment:
I think we should use the svg hint (similar to the check we do when rendering.

Reply:
For the hint we speculatively decode the content to check if it contains an SVG. Adding that would lead to decoding the same data twice and even if the parameter is sent in the widget we already know we have to decode the data so there is no point in that.
I get that it is cleaner, but it won't save us anything and will only complicate the code (but I can still do it if you want).